File: app/src/main/java/com/ytempest/wanandroid/base/fragment/LoaderFrag.java
package com.ytempest.wanandroid.base.fragment;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import android.view.View;
import android.view.ViewGroup;

import com.ytempest.wanandroid.base.load.Loader;
import com.ytempest.wanandroid.base.presenter.IPresenter;

/**
 * @author heqidu
 * @since 2020/12/28
 */
public class LoaderFrag<Presenter extends IPresenter> extends MvpFragment<Presenter> {

    private static final String TAG = LoaderFrag.class.getSimpleName();

    private Loader mLoader;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mLoader = new Loader((ViewGroup) view);
        mLoader.setReloadCall(aVoid -> onReloadClick());
    }

    protected void onReloadClick() {

    }

    public Loader getLoader() {
        return mLoader;
    }
}

File: app/src/main/java/com/ytempest/wanandroid/base/fragment/LoaderFrag.kt
package com.ytempest.wanandroid.base.fragment

import android.os.Bundle
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.ytempest.wanandroid.base.load.Loader
import com.ytempest.wanandroid.base.load.ViewType
import com.ytempest.wanandroid.base.presenter.IPresenter

/**
 * @author heqidu
 * @since 21-2-8
 */
abstract class LoaderFrag<Presenter : IPresenter, VB : ViewBinding> : MvpFragment<Presenter, VB>() {
    private lateinit var mLoader: Loader

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mLoader = Loader(view as ViewGroup)
        mLoader.setReloadCall {
            mLoader.showView(ViewType.LOAD)
            onReloadClick()
        }
    }

    protected open fun onReloadClick() {
    }

    fun getLoader(): Loader {
        return mLoader
    }
}